### PR TITLE
800 fix minor realization filter issues

### PR DESCRIPTION
**Pull request by jorgenherje**
Closes: #800

### CODE REVIEW

File: frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx [hunk 1]
```diff
@@ -77,19 +95,108 @@ export const RealizationFilterSettings: React.FC<RealizationFilterSettingsProps>
...
-            setActiveFilterEnsembleIdent(null);
+        ).length;
+        setNumberOfUnsavedRealizationFilters(numberOfUnsavedFilters);
+    }
+
+    const handleApplyAllClick = React.useCallback(
+        function handleApplyAllClick() {
+            // Apply all the unsaved changes state and reset the unsaved changes state
+            const resetHasUnsavedChangesMap: { [ensembleIdentString: string]: boolean } = {};
+            for (const ensembleIdentString of Object.keys(ensembleIdentStringToRealizationFilterSelectionsMap)) {
```
Comment (Anders2303) [inline frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx:110]: I you want to iterate the keys, you can simplify this by using a [`for...in` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)

File: frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx [hunk 2]
```diff
@@ -215,10 +276,17 @@ export const RealizationFilterSettings: React.FC<RealizationFilterSettingsProps>
...
             [ensembleIdentString]: hasUnsavedChanges,
-        });
+        };
+        setEnsembleIdentStringHasUnsavedChangesMap(newHasUnsavedChangesMap);
+
+        // Update number of unsaved filters
+        const numberOfUnsavedFilters = Object.values(newHasUnsavedChangesMap).filter(
+            (hasUnsavedChanges) => hasUnsavedChanges
+        ).length;
+        setNumberOfUnsavedRealizationFilters(numberOfUnsavedFilters);
```
Comment (Anders2303) [inline frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx:289]: You've repeated these lines a bunch of times. Could be nice to extract it into a utility function for it

File: frontend/src/framework/internal/components/RightSettingsPanel/rightSettingsPanel.tsx [hunk 3]
```diff
@@ -1,31 +1,83 @@
...
+        setDialogOpen(false);
+    }
+
     return (
         <div className="bg-white border-r-2 z-50 flex flex-col w-full h-full">
-            <RealizationFilterSettings workbench={props.workbench} onClose={handleRealizationFilterSettingsClose} />
-            <ModuleInstanceLog workbench={props.workbench} onClose={handleModuleInstanceLogClose} />
+            <RealizationFilterSettings workbench={props.workbench} onClose={handleOnClose} />
+            <ModuleInstanceLog workbench={props.workbench} onClose={handleOnClose} />
+            {
```
Comment (Anders2303) [inline frontend/src/framework/internal/components/RightSettingsPanel/rightSettingsPanel.tsx:59]: These curly brackets are not supposed to be here, right?

**Review by Anders2303 — APPROVED** [review]
Functionally it seems fine to merge.

I found some small code quality things that migth want to be changed before merging, but purely optional

File: frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx [hunk 4]
```diff
@@ -37,6 +48,13 @@ export const RealizationFilterSettings: React.FC<RealizationFilterSettingsProps>
         [ensembleIdentString: string]: EnsembleRealizationFilterSelections;
     }>({});
 
+    // Set no active filter if the settings panel is closed
+    let actualActiveFilterEnsembleIdent = activeFilterEnsembleIdent;
+    if (rightSettingsPanelWidth < 5 && activeFilterEnsembleIdent !== null) {
+        setActiveFilterEnsembleIdent(null);
+        actualActiveFilterEnsembleIdent = null;
+    }
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply frontend/src/framework/internal/components/RightSettingsPanel/private-components/RealizationFilterSettings/realizationFilterSettings.tsx:56]: Sure, in reality it should not affect the functionality.

The thought was to have correct state at first render after rightSettingsPanelWidth < 5, as a state value is not updated before next render of the component. Which implies an activeFilterEnsembleIdent on first render after a closed settings panel/drawer on the right side.

File: frontend/src/framework/internal/components/EnsembleRealizationFilter/ensembleRealizationFilter.tsx [hunk 5]
```diff
@@ -213,31 +213,40 @@ export const EnsembleRealizationFilter: React.FC<EnsembleRealizationFilterProps>
...
+                "ring-2 ring-opacity-100": !props.isActive && !props.isAnotherFilterActive,
+                "ring-orange-400 shadow-orange-400": props.hasUnsavedSelections,
+                "ring-blue-400 shadow-blue-400": !props.hasUnsavedSelections && props.isActive,
+                "ring-gray-300 shadow-gray-300 ": !props.hasUnsavedSelections && !props.isActive,
+                "hover:ring-blue-400 hover:shadow-blue-400 hover:shadow-md":
+                    !props.hasUnsavedSelections && !props.isActive && !props.isAnotherFilterActive,
+                "hover:shadow-blue-400 hover:shadow-lg shadow-md":
+                    props.hasUnsavedSelections && !props.isActive && !props.isAnotherFilterActive,
+                "group hover:shadow-md hover:ring-opacity-75 transition-opacity duration-100":
+                    !props.isActive && props.isAnotherFilterActive,
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply frontend/src/framework/internal/components/EnsembleRealizationFilter/ensembleRealizationFilter.tsx:229]: Valid point.

**Pull request merged**